metaflow/plugins/conda_escape/__init__.py

```diff
@@ -0,0 +1,162 @@
+# Portions of this plugin are inspired by RPyC
+# https://rpyc.readthedocs.io/
+#
+# The license for that software is reproduced below
+#
+# Copyright (c) 2005-2013
+#  Tomer Filiba (tomerfiliba@gmail.com)
+#  Copyrights of patches are held by their respective submitters
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os
+import sys
+from subprocess import Popen, PIPE
+
+from itertools import chain
+
+from .exception_transferer import RemoteInterpreterException
+from .client_modules import create_modules
+
+# Determine what is the python executable to use for the escape hatch. To do this,
+# we look for ESCAPE_HATCH_PY in the environment AND STORE IT. When metaflow
+# is first launched by the user, it is in the outside non-metaflow-created-conda
+# environment and we will consider that to be the environment we escape to.
+# Unless the user wants to override it, this will be unset and we will use
+# the Python we are running in. We then store it so that children run (*within*
+# the conda environment) use this same value
+ESCAPE_HATCH_PY = os.environ.get('METAFLOW_ESCAPE_HATCH_PY', sys.executable)
+os.environ['METAFLOW_ESCAPE_HATCH_PY'] = ESCAPE_HATCH_PY
+
+
+def generate_trampolines(python_interpreter_path, python_path):
+    # This function will look in the configurations directory and create
+    # files named <module>.py that will properly setup the escape hatch when
+    # called
+
+    # in some cases we may want to disable escape hatch
+    # functionality, in that case, set MF_ESCAPE_HATCH_DISABLED
+    if os.environ.get('METAFLOW_ESCAPE_HATCH_DISABLED', False) in (True, 'True'):
+        return
+
+    if python_interpreter_path is None:
+        python_interpreter_path = ''
+
+    paths = [os.path.dirname(os.path.abspath(__file__)) + "/configurations"]
+    try:
+        import metaflow_custom.plugins.conda_escape as custom_escape
```

Comment:
Similar comment as the top of the file - maybe we can remove references to `conda` in the directory.